src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -165,10 +207,33 @@ private String createOrGetCollection() throws InterruptedException {
         return collectionName;
     }
 
-    private void seedIndex(List<String> vectorIds, String indexName) throws InterruptedException {
+    private void seedDefaultNamespace(String indexName) throws InterruptedException {
         // Build upsert request
         Index indexClient = pineconeClient.getIndexConnection(indexName);
-        indexClient.upsert(buildRequiredUpsertRequestByDimension(vectorIds, dimension), "");
+
+        // Upsert vectors into both default namespace
+        indexClient.upsert(buildRequiredUpsertRequestByDimension(vectorIdsForDefaultNamespace, dimension), defaultNamespace);
+
+        // Wait for record freshness
+        DescribeIndexStatsResponse indexStats = indexClient.describeIndexStats();
+
+        int totalTimeWaitedForVectors = 0;
+        while (indexStats.getTotalVectorCount() == 0 || totalTimeWaitedForVectors <= 60000) {
+            Thread.sleep(2000);
+            totalTimeWaitedForVectors += 2000;
+            indexStats = indexClient.describeIndexStats();
+        }
+        if (indexStats.getTotalVectorCount() == 0 && totalTimeWaitedForVectors >= 60000) {
+            throw new PineconeException("Failed to seed index " + indexName + "with vectors");
+        }
+    }
+
+    private void seedCustomNamespace(String indexName) throws InterruptedException {
+        // Build upsert request
+        Index indexClient = pineconeClient.getIndexConnection(indexName);
+
+        // Upsert vectors into both custom namespace
+        indexClient.upsert(buildRequiredUpsertRequestByDimension(vectorIdsForCustomNamespace, dimension), customNamespace);
 
```

Comment:
Since the seed methods are largely the same, it seems like they could just be the same method that takes a namespace and list of vectors as input.

Reply:
Ahhh true !